packages/squiggle-lang/src/cli/commands/run.tsx

```diff
@@ -0,0 +1,290 @@
+        <ModuleDataInfo
+          key={hash}
+          hash={hash}
+          state={state}
+          moduleData={moduleData}
+          environment={environment}
+        />
+      ))}
+    </Box>
+  );
+};
+
+async function _run(
+  args: Pick<
+    RunArgs,
+    | "src"
+    | "filename"
+    | "runner"
+    | "runnerThreads"
+    | "showProjectState"
+    | "logProjectActions"
+  > & {
+    environment: Env;
+  }
+) {
+  const linker = getLinker();
+  const runner = args.runner
+    ? runnerByName(args.runner, args.runnerThreads ?? 1)
+    : undefined;
+
+  const project = new SqProject({
+    linker,
+    runner,
+    environment: args.environment,
+  });
+
+  const rootSource = new SqModule({
+    name: args.filename ?? EVAL_SOURCE_ID,
+    code: args.src,
+  });
+
+  const showState = () => {
+    render(<StateGraph state={project.state} environment={args.environment} />);
+  };
+
+  if (args.showProjectState) {
+    project.addEventListener("action", showState);
+  }
+
+  if (args.logProjectActions) {
+    project.addEventListener("action", (action) => {
+      console.log(action.data);
+    });
+  }
+
+  project.setHead("root", { module: rootSource });
+  return await project.waitForOutput("root");
+}
+
+async function run(args: RunArgs) {
+  const environment: Env = {
+    ...(args.sampleCount && isFinite(Number(args.sampleCount))
+      ? {
+          sampleCount: Number(args.sampleCount),
+          xyPointLength: Number(args.sampleCount),
+        }
+      : {
+          sampleCount: defaultEnv.sampleCount,
+          xyPointLength: defaultEnv.xyPointLength,
+        }),
+    seed: args.seed || "default-seed",
+    profile: args.profile ?? false,
+  };
+
+  const output = await _run({
+    src: args.src,
+    filename: args.filename,
+    environment,
+    runner: args.runner,
+    runnerThreads: args.runnerThreads,
+    showProjectState: args.showProjectState,
+    logProjectActions: args.logProjectActions,
+  });
+
+  const printer = new CliPrinter();
+  if (!output.result.ok) {
+    printer.printSection(
+      red("Error:"),
+      output.result.value.toStringWithDetails()
+    );
+  } else {
+    const outputResult = output.result.value;
+    switch (args.output) {
+      case "RESULT_OR_BINDINGS":
+        if (outputResult.result.tag === "Void") {
+          printer.printSection(outputResult.bindings.toString());
+        } else {
+          printer.printSection(outputResult.result.toString());
+        }
+        break;
+      case "RESULT_AND_BINDINGS":
+        printer.printSection(bold("Result:"), outputResult.result.toString());
+        printer.printSection(
+          bold("Bindings:"),
+          outputResult.bindings.toString()
+        );
+        break;
+      case "NONE":
+      // do nothing
+    }
+  }
+
+  if (args.measure) {
+    printer.printSection(`${bold("Time:")} ${output.executionTime}s`);
+  }
+}
+
+export function addRunCommand(program: Command) {
```

Comment:
Is this cli tool documented somewhere? Still seems like there should be some brief README.md file here.

I get that there's the website documentation, but that seems geared towards a different audience (more explaining the underlying system, not discussing as much the tldr of how to use it)

Reply:
The CLI tool is kind of self-documenting:

```
$ node dist/cli/index.js --help
Usage: index [options] [command]

Options:
  -h, --help                     display help for command

Commands:
  run [options] [filename]
  parse [options] [filename]
  print-ir [options] [filename]
  playground
  help [command]                 display help for command
$ node dist/cli/index.js run --help
Usage: index run [options] [filename]

Options:
  -e, --eval <code>          run a given squiggle code string instead of a file
  -t --time                  output the time it took to evaluate the code
  -p --profile               performance profiler
  -q, --quiet                don't output the results and bindings
  --show-project-state
  --log-project-actions
  -r, --runner <runner>      embedded (choices: "web-worker", "embedded", "embedded-with-serialization")
  --runner-threads <number>
  -b, --show-bindings        show bindings even if the result is present
  -h, --help                 display help for command
```

It can be improved, but that feels out of scope of this PR.

I wanted to mention this in the top-level `@quri/squiggle-lang` README, but it's so outdated that it's also out of scope.